core/src/main/java/com/novoda/noplayer/mediaplayer/AndroidMediaPlayerFacade.java

```diff
@@ -125,6 +126,7 @@ public void onPrepared(MediaPlayer mp) {
         @Override
         public void onCompletion(MediaPlayer mp) {
             currentState = PlaybackState.COMPLETED;
+            MediaPlayer.OnCompletionListener onCompletionForwarder = forwarder.onCompletionListener();
```

Comment:
If the question is:

`Can the player of the library force this listeners to be null`? The question is no, because in the worst case scenario the forwarder will have a no-op implementation contained. 

But it is possible that we, developers, by mistake do something that causes it hold a `null`. You can consider the thrown below as a `DeveloperError`